### ISSUE TITLE
music doesn't stop after disconnection a bluetooth headset

### ISSUE THREAD

**Issue by mymike00** [ubports/ubuntu-touch#979]
- Device: nexus 5
- Channel: rc
- Build: 2018-W48

### Steps to reproduce

Connect a bluetooth headset, open the music app and reproduce some music. then switch off the bluetooth headset.

### Expected behavior

I expect the music to stop, phone's speakers shouldn't play anything

### Actual behavior

the music continues playing through phone's speakers. that's the very first time it happens to me.
all the other times music were stopped when the bluetooth headset was turned off...

**Comment by Flohack74** [comment]
Try a wired headset, does it stop then?

**Comment by nathangiovannini3** [comment]
Device: nexus 5
    Channel: development 
    Build: 2018-12-08 

  
Also I noticed this problem with Bluetooth headphones. While with the headphones wired the music stops normally.

**Comment by mymike00** [comment]
The strange thing is that it doesn't happens always, I noticed it 2 times...
but with bluetooth in generals there are some problems ... some times with bluetooth headset connected and bluoetooth speakers written in the volume bar, the sound is produced via phobe speakers...

**Comment by hummlbach** [comment]
On turbo@ota7:
- when wired headphones are attached: audio playback is redirected to the headphones (no more audio on the speakers as expected)
- when wired headphones are detached: playback continues on the speaker
- when bluetooth headphones are detached: music continues playing on speaker
- when bluetooth headphones are attached: playback continues on phones speaker instead of switching to the headphones - app restart needed to get it on the headphones
(youtube app used)

**Comment by Flohack74** [comment]
@hummlbach are you sure the wired headphones don´t stop the playback? It actually worked for me like this most of the time.

**Comment by mymike00** [comment]
Yeah, wired HP stop the music for me too

**Comment by Flohack74** [comment]
I can´t reproduce this, tried a few times now with just switching off the BT headset. Music App always stopped, though you can hear a small hiccup when switching, coming from the speaker.

**Comment by mymike00** [comment]
Yeah, I hear that hiccup also with wired headphones. regarding bt headset I'll test them asap to see if this happens again or not

**Comment by UniversalSuperBox** [comment]
Hmm, I'm also unable to confirm this. Anything playing with the media player service stops when Bluetooth is disconnected.

However, https://github.com/ubports/ubuntu-touch/issues/1045 does state that, upon reconnecting, music does not switch back to the Bluetooth speaker. That issue should be tracked there.

**Comment by arubislander** [comment]
I can confirm this on my Meizu PRO 5. When bluetooth speakers are disconnected the sound continues playing on the device's speakers.

**Comment by mateosalta** [comment]
This is happening on my volla too, bluetooth looses connection then sound comes out through the phone, although only tested with "music" from youtube, and panpipe(pinepipe is using media hub) not sure about music app

**Comment by UniversalSuperBox** [comment]
This actually appears to be related to https://github.com/ubports/ubuntu-touch/issues/1415, which explains why I had difficulty confirming it. The issue only occurs if Bluetooth headphones are disconnected while the application that requested the current audio's playback has been suspended during a track change. If the application is in the foreground when the disconnection occurs, or the playing track has not changed since the app was suspended, this issue does not occur.
